test/Microsoft.TemplateEngine.Cli.UnitTests/TemplateResolutionTests/ListTemplateListResolverTests.cs

```diff
@@ -741,5 +741,98 @@ public void TestGetTemplateResolutionResult_MatchByTagsAndMismatchByFilter()
             Assert.True(matchResult.HasContextMismatch);
             Assert.False(matchResult.HasBaselineMismatch);
         }
+
+
+        [Fact(DisplayName = nameof(TestGetTemplateResolutionResult_MatchByAuthor))]
+        public void TestGetTemplateResolutionResult_MatchByAuthor()
+        {
+            List<ITemplateInfo> templatesToSearch = new List<ITemplateInfo>();
+            templatesToSearch.Add(new TemplateInfo()
+            {
+                ShortName = "console",
+                Name = "Long name for Console App",
+                Identity = "Console.App.T1",
+                GroupIdentity = "Console.App.Test",
+                CacheParameters = new Dictionary<string, ICacheParameter>(),
+                Classifications = new List<string> { "Common", "Test" },
+                Author = "TestAuthor",
+                Tags = new Dictionary<string, ICacheTag>(StringComparer.OrdinalIgnoreCase)
+                {
+                    { "language", ResolutionTestHelper.CreateTestCacheTag("L1") },
+                    { "type", ResolutionTestHelper.CreateTestCacheTag("project")}
+                },
+                BaselineInfo = new Dictionary<string, IBaselineInfo>()
+                {
+                    { "app", new BaselineInfo() },
+                    { "standard", new BaselineInfo() }
+                }
+            });
+
+
+            INewCommandInput userInputs = new MockNewCommandInput()
+            {
+                TemplateName = "console",
+                IsListFlagSpecified = true,
+                AuthorFilter = "Test"
+            };
+
+            ListOrHelpTemplateListResolutionResult matchResult = TemplateListResolver.GetTemplateResolutionResultForListOrHelp(templatesToSearch, new MockHostSpecificDataLoader(), userInputs, null);
+            Assert.True(matchResult.HasExactMatches);
+            Assert.Equal(1, matchResult.ExactMatchedTemplatesGrouped.Count);
+            Assert.Equal(1, matchResult.ExactMatchedTemplates.Count);
+            Assert.False(matchResult.HasPartialMatches);
+            Assert.Equal(0, matchResult.PartiallyMatchedTemplates.Count);
+            Assert.Equal(0, matchResult.PartiallyMatchedTemplatesGrouped.Count);
+            Assert.False(matchResult.HasLanguageMismatch);
+            Assert.False(matchResult.HasContextMismatch);
+            Assert.False(matchResult.HasBaselineMismatch);
+            Assert.False(matchResult.HasAuthorMismatch);
+        }
+
+        [Fact(DisplayName = nameof(TestGetTemplateResolutionResult_MismatchByAuthor))]
+        public void TestGetTemplateResolutionResult_MismatchByAuthor()
```

Comment:
I don't know how it can be done in Xunit, but it would be nice if we could supply multiple test inputs to the same test function similar to what we did here https://github.com/dotnet/insertions-client/blob/0beae3a0fb1d8c238bc79c535fe41258af4d8275/tests/InsertionsClient.Console.Test/BuildFilterFactoryTests.cs#L27 .

We could test different author names and inputs easily. Some things I think would be good to test (some of these should succeed, some should fail):

| Author | input |
| ---- | ---- |
| testauthor |  |
| testauthor | [null] |
| test |  TeSt |
| tester |  Teşt |
| input | İnput |
| match_middle_test | middle |